app/services/activitypub/fetch_remote_status_service.rb

```diff
@@ -80,4 +80,20 @@ def expected_activity_type?
   def expected_object_type?
     equals_or_includes_any?(@json['type'], ActivityPub::Activity::Create::SUPPORTED_TYPES + ActivityPub::Activity::Create::CONVERTED_TYPES)
   end
+
+  def fetch_status(uri, id_is_known, on_behalf_of = nil)
+    begin
+      fetch_resource(uri, id_is_known, on_behalf_of, true)
+    rescue Mastodon::UnexpectedResponseError => e
+      return unless e.response.code == 404
+
+      # If this is a 404 from a status from an account that has no local followers, delete it
+      existing_status = Status.find_by(uri: uri)
+      if !existing_status.nil? && existing_status.unsubscribed?
+        Rails.logger.debug { "FetchRemoteStatusService - Got 404 for orphaned status with URI #{uri}, deleting" }
+        Tombstone.find_or_create_by(uri: uri, account: existing_status.account)
+        RemoveStatusService.new.call(existing_status, redraft: false)
```

Comment:
I didn't want to fake a `Delete` action and return it, since an explicit `Delete` should probably stay semantically different from this. I saw that `Activity::Delete` also makes a redis lock around these operations, but i wasn't sure if this was necessary here since this was internally initiated, and there is another redis lock just within the `RemoveStatusService::call` method